### DISCUSSION TITLE
I can't understand the axis_neuron value

### DISCUSSION THREAD

**Discussion by saeedshahi** [deepmodeling/deepmd-kit#1251] [Q&A] [answered]
Hello all,

I want to train water with different Embedding net sizes like [1,2,4], [5,10,20], and [15,30,60,120] but I don't know what value should I choose for  axis_neuron.
I found this description for that on the DeePMD website "The axis_neuron specifies the size of the submatrix of the embedding matrix, the axis matrix as explained in the DeepPot-SE paper" and I read DeepPot-SE paper but still I don't understand it.
Can you briefly describe that and if it is possible to give me a value for this net [1,2,4], [5,10,20], and [15,30,60,120], also is it an exact value or there is a range of numbers that we can use?

Thanks

**Comment by wanghan-iapcm** [comment, marked as answer]
we take a submatrix G_2 of the embedding matrix G_1 to construct the descriptor D = G_1^T R R^T G_2. The size of the submatrix is specified by the `axis_neuron` parameter.

**Reply by saeedshahi** [reply]
> we take a submatrix G_2 of the embedding matrix G_1 to construct the descriptor D = G_1^T R R^T G_2. The size of the submatrix is specified by the `axis_neuron` parameter.

Thanks, Dr. Wang for your prompt response.
How can determine the size of the G_2 matrix, as I understand it is a portion of the G_1 matrix and always the size of G_2 < G_1 but my question is that what is a limitation for choosing a submatrix from G_1? can we choose any portion and size of G_1 as long as G_2 < G_1? Has G_2 any physical meaning like a radial portion of a descriptor or it is just a mathematical trick?

Best,
Saeed

**Reply by wanghan-iapcm** [reply]
One may choose any size of submatrix G_2 from G_1. We have tested different settings with  `axis_neuron` >= 4 and find that the final accuracy is not sensitive to this parameter. A larger G_2 will definitely lead to more computational and memory cost. Thus we would recommend always setting   `axis_neuron` >= 4.
